cloudflare/src/framework/mod.rs

```diff
@@ -2,17 +2,20 @@
 This module controls how requests are sent to Cloudflare's API, and how responses are parsed from it.
  */
 pub mod apiclient;
+#[cfg(feature = "reqwest-http")]
```

Comment:
Self-comment: and this is what we wanted to achieve, we can now cleanly elide a few modules away and get the crate to compile under wasm.

Without the changes in this PR this would be a mess of macros all around due to the more promiscuous coupling of reqwest that was all around.
it's still not perfect, but good enough for now